agent/src/softwarecontaineragent.cpp

```diff
@@ -252,7 +293,8 @@ std::string SoftwareContainerAgent::bindMountFolderInContainer(const uint32_t co
     return "";
 }
 
-void SoftwareContainerAgent::setGatewayConfigs(const uint32_t &containerID, const std::map<std::string, std::string> &configs)
+void SoftwareContainerAgent::setGatewayConfigs(const uint32_t &containerID,
+                                               const GatewayConfiguration &configs)
```

Comment:
Yes, I wasn't sure about this, but I agree. What does @tobsan say about this change?